### ISSUE TITLE
Listen for the X icon action from the cast notification

### ISSUE THREAD

**Issue by julianCast** [jellyfin/cordova-plugin-chromecast#73]
Hi again! I'm almost done!
Here is the thing...I'm able to successfully listen to the event from the **STOP CASTING**  button using:

```
this._session.addUpdateListener(function listenerSession() {
   if (this._session.status === window.chrome.cast.SessionStatus.STOPPED) {  /* here *\ }
```

![photo5789456323069457380](https://user-images.githubusercontent.com/559677/96335258-ba703300-1077-11eb-91cc-7ddaf3bfdaf2.jpg)

But I'm not able when I touch the **X** in the cast notification. As far as I understand that stop the session too.

![photo5789456323069457381](https://user-images.githubusercontent.com/559677/96335257-b9d79c80-1077-11eb-860a-e158761dff03.jpg)

Thanks !!

**Comment by Lindsay-Needs-Sleep** [comment]
Yep, I agree that it **should** also trigger that event with `SessionStatus.STOPPED`

This was not something I tested.

The second picture, is that from an Android home screen?  Or the locked screen?  
(All of my test devices have their lock screens disabled so I never encountered this UI.  Do you have to do anything special to enable it?)

If it is on the locked screen, it might be a bit difficult to get the event triggered (since the app might not be running).  Not sure.

**Comment by julianCast** [comment]
That wasn't the locks creen, just the notification when I slide down the screen (having the app open behind)

The pause icon does trigger its event.

**Comment by Lindsay-Needs-Sleep** [comment]
Ooh okay!

Well it **should** be fairly easy to get that work properly then I think/hope.  

(I’m guessing the close event is being ignored on the native side for some reason.  Or maybe the notification close/stop casting event needs a different listener.)

I won’t have time for quite awhile (3 weeks+) to look at this.

If you are interested in fixing this, I can point you at code sections that I think would be relevant.

**Comment by julianCast** [comment]
Oh ok! Yes, I could take a look in the meantime, can provide me some hints? Thanks!

**Comment by Lindsay-Needs-Sleep** [comment]
Yep!

**I think your best bet is about here:**
https://github.com/jellyfin/cordova-plugin-chromecast/blob/master/src/android/ChromecastSession.java#L129

In particular:
```java
                    @Override
                    public void onApplicationDisconnected(int i) {
                        clientListener.onSessionEnd(
                                ChromecastUtilities.createSessionObject(session, "stopped"));
                    }
```
My guess is that this event isn't being triggered.  
Either:  
* there is another event that needs to be listened to in `session.addCastListener(new Cast.Listener() {`  
    * Check documentation for `Cast.listener`
* Or `session.addCastListener(new Cast.Listener() {` is not getting called in time.  (I doubt it is this.)

`clientListener.onSessionEnd(ChromecastUtilities.createSessionObject(session, "stopped"));` is the part that actually results in calling/triggering the javascript event.  So you just need to figure out where this additionally needs to be called.

**Comment by Lindsay-Needs-Sleep** [comment]
I am working on some chromecast stuff now, @julianCast if you have had any luck with figuring this out, please let me know!  (I will hopefully be able to test and add it)

**Comment by julianCast** [comment]
I am right now looking into it, but so far is not something obvious, the `session.addCastListener` is listening all the events availables, except for [onActiveInputStateChanged](https://developers.google.com/android/reference/com/google/android/gms/cast/Cast.Listener?hl=es#onActiveInputStateChanged(int)), but that is not related to stop session.

**Comment by julianCast** [comment]
Debuggin here: https://github.com/jellyfin/cordova-plugin-chromecast/blob/master/www/chrome.cast.js#L1439

When I quit the session within the app I get as eventname:
RECEIVER_LISTENER then SESSION_UPDATE (which is the one that trigger the event stopped)

When I quit the session using the notification I do get something, but only:
RECEIVER LISTENER

**Comment by Lindsay-Needs-Sleep** [comment]
(I finally got around to try it myself, can confirm that it doesn't work right)

Interesting, so it looks like it is half working. :/
Thanks for checking `session.addCastListener`! too bad that wasn't the answer. :/

If you find anything else please let me know!  (I am on a very tight time schedule so I'm not sure if I will be able to get to it on my own)
